docs/current_docs/quickstart/agent/part2.mdx

```diff
@@ -0,0 +1,377 @@
+
+    ```php file=./snippets/part3/agent/php/agent.php.snippet
+    ```
+
+    Be sure to add imports for the following:
+
+    ```php
+    use Dagger\Secret;
+    ```
+    </TabItem>
+    <TabItem value="java" label="Java">
+    Add the following new function to the main dagger module in `.dagger/src/main/java/io/dagger/modules/hellodagger/HelloDagger.java`:
+
+    ```java file=./snippets/part3/agent/java/agent.java.snippet
+    ```
+
+    Be sure to add imports for the following:
+
+    ```java
+import io.dagger.client.Secret;
+import io.dagger.client.GithubIssue;
+    ```
+    </TabItem>
+</Tabs>
+
+The `develop-issue` function connects the dots in the automation flow. It will be called when a Github issue is ready to be developed. Using the `github-issue` dependency, this code will:
+
+- Read the title and body from the Github issue
+- Use the body as the assignment for the agent
+- Open a pull request with the Directory returned by the agent
+
+### Create a Github Actions workflow
+
+Github Actions will need the ability to call out to an LLM. It will need the same environment variables you [configured](../../configuration/llm.mdx) locally setup as repository secrets.
+
+To configure new repository secrets:
+- From your Github repository, select the Settings tab
+- Under Security, expand the section "Secrets and Variables" and select Actions
+- Select "New repository secret"
+- Create a secret for each variable required for your LLM provider. For example: `GEMINI_API_KEY`
+
+![Configure Github Actions repository secrets](/img/current_docs/quickstart/agent/configure_gha_llm.png)
+
+If you signed up for Dagger Cloud, you should also configure a `DAGGER_CLOUD_TOKEN` repository secret to see your Github Actions traces in Dagger Cloud.
+
+To find your Dagger Cloud token, navigate to the organization settings page using the cogwheel icon in the top navigation bar. Under the `Tokens` sub-menu, click the eye icon to view the token. You can also use this URL pattern: `https://v3.dagger.cloud/{Your Org Name}/settings?tab=Tokens`
+
+![Get token](/img/current_docs/configuration/get-token.png)
+
+The repository also needs to be configured to give Github Actions the ability to create pull requests. On the left side, select Actions -> General and find the option "Allow GitHub Actions to create and approve pull requests". Make sure this is enabled.
+
+![Allow Github Actions to create pull requests](/img/current_docs/quickstart/agent/configure_gha_prs.png)
+
+Create the file `.github/workflows/develop.yml` with the following content:
+
+```yaml file=./snippets/part3/develop.yml
+```
+
+This snippet shows an example with Google Gemini by specifying `GEMINI_API_KEY`. Change this to match the repository secrets you created earlier.
+
+This Github Actions workflow will call the `develop-issue` function when the label "develop" is added to an issue. The Github token passed to the agent has permissions to read issues, write contents, and write pull-requests.
```

Comment:
```suggestion
This GitHub Actions workflow will call the `develop-issue` function when the label "develop" is added to an issue. The GitHub token passed to the agent has permissions to read issues, write contents, and create pull requests.
```

Reply:
The word "write" is used instead of "create" because the token permissions are either "read" or "write" and the token has "write" for pull requests

Reply:
Ah OK, thanks for clarifying! Would it make sense to have those verbs in code font to make it clear they refer to actual permissions?